src/solvers/smt2/smt2irep.cpp

```diff
@@ -47,7 +47,13 @@ bool smt2irept::parse()
       switch(next_token())
       {
       case END_OF_FILE:
-        throw error("unexpected end of file");
+        if(stack.empty())
+        {
+          result = irept(irep_idt());
```

Comment:
Is this really necessary with the `result.clear()` above?

Reply:
No, not strictly; this is primarily meant to make it clear what is returned.

Reply:
I guess to really make it clear the best way would be to make `parse` return an `optionalt<irept>` instead of a bool.

Reply:
+1 for using `optionalt<irept>`

Reply:
But then also this commit should just be in a PR of its own where those kinds of changes could be made. It really doesn't relate to floating-point encoding fixes in any obvious way.